### ISSUE TITLE
False-positve `UnusedImports` when there are mutiple functions of the same name

### ISSUE THREAD

**Issue by sschuberth** [detekt/detekt#7224]
## Expected Behavior

Detekt should not report `UnusedImports` for imports that are actually required.

## Observed Behavior

Detekt flags an import that is required as `UnusedImports`.

## Steps to Reproduce

In this code

```
import org.eclipse.apoapsis.ortserver.dao.createDataSource

private fun Application.createDataSource(): DataSource {
    val configManager: ConfigManager by inject()

    val dataSourceConfig = DatabaseConfig.create(configManager)

    return createDataSource(dataSourceConfig)
}
```

there is both the imported `createDataSource` (non-extension) function, and a function of the same name in the same file. Needless to say that `createDataSource(dataSourceConfig)` cannot call `Application.createDataSource()` as the former has an argument provided, but the latter takes none. So it's clear that only `org.eclipse.apoapsis.ortserver.dao.createDataSource` can be meant, and has to be imported.

## Context

I'm trying to resolve all Detekt hints in my code base, using type resolution.

## Your Environment

* Version of detekt used: 1.23.6
* Version of Gradle used (if applicable): 8.7
* Gradle scan link (add `--scan` option when running the gradle task): n/a
* Operating System and version: Fedora Linux 39
* Link to your project (if it's a public repository): https://github.com/eclipse-apoapsis/ort-server/blob/main/core/src/main/kotlin/plugins/Database.kt

**Comment by atulgpt** [comment]
Hi @sschuberth I tried below TC in the latest main and it is passing. 

```kotlin
@Test
fun `does not report when normal function is imported in file containing extension function with the same name - #7224`() {
    val mainFile = """
        import java.math.BigDecimal
        
        private fun BigDecimal.createBigDecimal(): BigDecimal {
            println("this = $this")
            return BigDecimal(9)
        }
    """.trimIndent()

    assertThat(subject.lintWithContext(env, mainFile)).isEmpty()
}
```

can you create minimal reproducible project or provide a failing TC?

**Comment by atulgpt** [comment]
Hi @sschuberth I tried with the below TC and it is passing

```kotlin
@Test
fun `does not report when normal function is imported in file containing extension function with the same name - #7224`() {
    val mainFile = """
        import kotlin.collections.listOf
        
        private fun Int.listOf(): List<Int> {
            return listOf(this)
        }
    """.trimIndent()

    assertThat(subject.lintWithContext(env, mainFile)).isEmpty()
}
```

Can you check if the TC is what you are trying to do?

**Comment by sschuberth** [comment]
I also cannot reproduce this one anymore... I wonder whether the migration of our project to Kotlin 2.0.0 made a difference here.
